src/client.ts

```diff
@@ -357,6 +424,15 @@ export default class Client {
     target: Target,
     defaultValue = {},
   ): Promise<Record<string, unknown>> {
+    if (!this.initialized) {
+      warnDefaultVariationServed(
+        identifier,
+        target,
+        defaultValue.toString(),
+        this.log,
+      );
+      Promise.resolve(defaultValue);
```

Comment:
I think you'll need to return this. As-is this `Promise` looks to be resolving into the void and the `evaluator.jsonVariation` will still get called.

Reply:
Same with the other ones above too

Reply:
Oh and super minor thing, but worth doing is to make sure conditions and returns have whitespace around them to make sure the file can be easily visually scanned. A wall of code makes it difficult to spot things.

Reply:
Done on both counts